controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -305,6 +318,249 @@ func (r *AutoscalingRunnerSetReconciler) deleteEphemeralRunnerSets(ctx context.C
 	return nil
 }
 
+func (r *AutoscalingRunnerSetReconciler) cleanupPermissions(ctx context.Context, autoscalingRunnerSet *v1alpha1.AutoscalingRunnerSet, logger logr.Logger) (requeue bool, err error) {
+	switch autoscalingRunnerSet.Annotations[autoscalingRunnerSetModeAnnotationKey] {
+	case "kubernetes":
+		requeue, err := r.cleanupContainerModeKubernetesPermissions(ctx, autoscalingRunnerSet, logger)
+		if err != nil {
+			return true, err
+		}
+		if requeue {
+			return true, nil
+		}
+	default:
+		if serviceAccountName, ok := autoscalingRunnerSet.Annotations[AnnotationKeyNoPermissionServiceAccountName]; ok {
+			logger.Info("Cleaning up no permission service account")
+			serviceAccount := new(corev1.ServiceAccount)
+			err = r.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: autoscalingRunnerSet.Namespace}, serviceAccount)
+			switch {
+			case err == nil:
+				if serviceAccount.DeletionTimestamp.IsZero() {
+					err := r.Delete(ctx, serviceAccount)
```

Comment:
delete or remove finalizer

Reply:
But also, what should we do if we get to this point and the deletion timestamp is zero? Removing finalizer is of no use since the resource will still exist. And since we know that this `if` will never evaluate to true if used by supported helm charts, it does not hurt to have extra logic here. At least, if someone is using a different way of applying our resources, it can signal that the permission is missing to clean up.

Reply:
My thought is whoever creates the resource is responsible for its deletion. 
Removing the finalizer basic means the AutoScalingRunnerSet gets deleted no longer taking a lock on those resources.

Reply:
Oh I see, removing the delete and changing the name to explain the call correctly :relaxed: